app/server/utils/user.py

```diff
@@ -592,6 +592,10 @@ def proccess_create_or_modify_user_request(
             response_object = {'message': 'User already exists for Identifier'}
             return response_object, 400
 
+        if (existing_user.id != user_id) and existing_user.phone == phone:
+            response_object = {'message': f'User already exists for phone {existing_user}'}
+            return response_object, 400
+
```

Comment:
Right now, if you try to update a users phone to an existing user phone, it will just silently fail